lib/tests/streamlit/runtime/session_manager_test.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from unittest.mock import MagicMock, patch
+
+import pytest
+from parameterized import parameterized
+
+from streamlit.runtime.session_manager import SessionManager
+
+
+class UnimplementedSessionManager(SessionManager):
+    pass
+
+
+class MinimalSessionManager(SessionManager):
+    def __init__(self, _session_storage, _runtime):
+        pass
+
+    def connect_session(self, _client, _user_info, _existing_session_id):
+        return "my_session_id"
+
+    def close_session(self, _session_id):
+        pass
+
+    def get_session_info(self, _session_id):
+        return None
+
+    def list_sessions(self):
+        return []
+
+
+class SessionManagerProtocolTests(unittest.TestCase):
```

Comment:
I might be misunderstanding, but this set of tests seems unnecessary to me - it seems to just be testing that class inheritance works in Python? (Or maybe it's just testing that someone doesn't accidentally remove the implementation of a formerly-implemented base class method? But still seems unusual/unnecessary for us to have an explicit test for this...?)

Shouldn't this stuff all be caught in the (eventual) tests that will actually use the SessionManager?

Reply:
(It _may_ be worthwhile to test the default implementation of these functions, but then we should have tests that explicitly test each of them)

Reply:
Hm, yeah. My intention was to test the default implementations are correct, but I think it'd be more sensible to just allow tests in concrete implementations to to do this when necessary.